website/docs/r/nsxt_edgegateway_dns.html.markdown

```diff
@@ -0,0 +1,133 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_nsxt_edgegateway_dns"
+sidebar_current: "docs-vcd-resource-nsxt-edgegateway-dns"
+description: |-
+  Provides a resource to manage NSX-T Edge Gateway DNS configuration.
+---
+
+# vcd\_nsxt\_edgegateway\_dns
+
+Supported in provider *v3.11+* and VCD 10.4.0+ with NSX-T.
+
+Provides a resource to manage NSX-T Edge Gateway DNS configuration.
+
+## Example Usage
+
+```hcl
+data "vcd_org_vdc" "v1" {
+  org  = "datacloud"
+  name = "nsxt-vdc-datacloud"
+}
+
+data "vcd_nsxt_edgegateway" "testing-in-vdc" {
+  org      = "datacloud"
+  owner_id = data.vcd_org_vdc.v1.id
+
+  name = "nsxt-gw-datacloud"
+}
+
+resource "vcd_nsxt_edgegateway_dns" "testing-in-vdc" {
+  org             = "datacloud"
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing-in-vdc.id
+  enabled         = true
+
+  default_forwarder_zone {
+    name = "testing"
+
+    upstream_servers = [
+      "1.1.1.1",
+      "2.2.2.2",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "conditional_testing"
+
+    upstream_servers = [
+      "3.3.3.3",
+      "4.4.4.4",
+    ]
+
+    domain_names = [
+      "example.org",
+      "nonexistent.org",
+    ]
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) Org in which the NSX-T Edge Gateway is located, required
+  if not set in the provider section.
+* `edge_gateway_id` - (Required) NSX-T Edge Gateway ID.
+* `enabled` - (Optional) Status of the DNS forwarding service. Defaults to `true`.
+* `listener_ip` - (Optional) The IP on which the DNS forwarder listens. If the Edge Gateway 
+  has a dedicated external network, this can be changed.
+* `snat_rule_ip_address` - (Optional, VCD 10.5.0+) This property only applies if the Edge Gateway 
+  is connected to a Provider Gateway using IP Spaces. If specified, VCD will 
+  conveniently manage the SNAT rule with the specified IP address for the DNS forwarder. 
+  The specified IP can be allocated using [`vcd_ip_space_ip_allocation`](/providers/vmware/vcd/latest/docs/resources/ip_space_ip_allocation) 
```

Comment:
I think it is worth documenting that it will populate (work as computed attribute) when not specified.

Reply:
Agree, I added `If not specified, the IP address will be computed and chosen by VCD.` to the argument, not sure if that makes it clear enough as it's tough to put it into not too many words.

Reply:
<img width="664" alt="image" src="https://github.com/vmware/terraform-provider-vcd/assets/15804230/f1faac34-ef8e-4aa7-85a8-b610a5d82dfd">

I would probably add a tiny highlight on the second case so that it is not completely ignored.

Something like
<img width="661" alt="image" src="https://github.com/vmware/terraform-provider-vcd/assets/15804230/86a04e81-a8c4-46f0-a052-e5c3b6d95ebe">

But that one is up to you